### PR TITLE
Adding in the chess competitions

### PR DESCRIPTION
**Pull request by AnUnsocialPigeon**
Also just sorting out the general competition area of the website. 

Also, the chess preview runs client-side, with server-side async data retreival from the /private data store. If this is not wise, then, LMK. I can fix.

I did kinda want it to be that the page gets served with the data already pre-prepared, however, given that there is an initial loading state, it seems that the website is sending a static pre-rendered version of the site, then supplying with the FEN data from the async serverside function. I am unsure how it is really working under-the-hood, as I can't see any specific packets in the network that have the FEN data in after the webpage is rendered. Oh well!

Pretty proud of the preview, also I think I have figured out how to fix the size issues that the chess board has on different devices ;)

### CODE REVIEW

**Review by AnUnsocialPigeon — COMMENTED** [review]
I have not ported over the courses page to match the ai_safety's implementation of the simple event. This should be done at some point // next time it is updated :)

File: src/app/competition/chess/page.tsx [hunk 1]
```diff
@@ -1,17 +1,138 @@
...
+// export default function Home() {
+//     return defaultPage(
+//         <>
+//             <Suspense fallback={<div>Loading...</div>}>
+//                 <ChessBoardPlayable />
+//             </Suspense>
+//             <BackButton />
+//         </>
+//     );
+// }
```
Comment (JakubCzarlinski) [inline src/app/competition/chess/page.tsx:138]: remove graveyard

File: src/classes/event_simple.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                    location: event.location,
+                    url: undefined,
+                    description: event.description
+                }
+                return calenderEvent;
+            }),
+            pictureUrl: event.pictureUrl,
+            simpleDate: event.simpleDate,
+        };
+    });
```
Comment (JakubCzarlinski) [inline src/classes/event_simple.tsx:45]: memory allocations go brrrrrrrrr

**Review by JakubCzarlinski — APPROVED** [review]
I think the auto linter isn't working, oh well... Can look into that another time

Watch out for those `.map`s - it's me being pedantic as a result of other projects though so you don't neeeed to replace them.

LGTM

**Comment by AnUnsocialPigeon** [comment]
Will clean up graveyard, and reduce some maps. Love me LinQ like structures but it breaks my heart that they're bad :(

File: src/app/competition/chess/page.tsx (hunk 1 above)
Comment (AnUnsocialPigeon) [reply src/app/competition/chess/page.tsx:138]: Straight forgot this was here. Sorry!

**Pull request merged**